bootloader/src/pyi_archive.c

```diff
@@ -449,6 +449,15 @@ int
 pyi_arch_set_paths(ARCHIVE_STATUS *status, char const * archivePath,
                    char const * archiveName)
 {
+    size_t pathlen, namelen;
+
+    pathlen = strnlen(archivePath, PATH_MAX);
```

Comment:
I noticed this, its due too waht I mentioned in the second part of [my last comment](https://github.com/pyinstaller/pyinstaller/pull/2503#issuecomment-286285855).
In short, I did not assume this POSIX standard was used.